src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,179 +1,280 @@
-
-                if (nonPugPattern.containsMatchIn(note)) {
-                    slot highlight LorenzColor.LIGHT_PURPLE
-                    continue
+            if (partyFinderTitlePattern.matches(inventoryName)) {
+                inInventory = true
+                for ((slot, stack) in event.inventoryItems) {
+                    val classNames = mutableListOf("Healer", "Mage", "Berserk", "Archer", "Tank")
+                    val toolTip = stack.getLore().toMutableList()
+                    for ((index, line) in stack.getLore().withIndex()) {
+                        classLevelPattern.matchMatcher(line) {
+                            val playerName = group("playerName")
+                            val className = group("className")
+                            val level = group("level").toInt()
+                            val color = getColor(level)
+                            if (config.coloredClassLevel) toolTip[index] = " §b$playerName§f: §e$className $color$level"
+                            classNames.remove(className)
+                        }
+                    }
+                    if (config.showMissingClasses && dungeonFloorPattern.matches(
+                            stack.getLore().firstOrNull()?.removeColor()
+                        )
+                    ) {
+                        if (classNames.contains(selectedClass)) classNames[classNames.indexOf(selectedClass)] =
+                            "§a${selectedClass}§7"
+                        toolTip.add("")
+                        toolTip.add("§cMissing: §7" + StringUtils.createCommaSeparatedList(classNames))
+                    }
+                    if (toolTip.isNotEmpty()) toolTipMap[slot] = toolTip
                 }
             }
+        }
 
-            val members = slot.stack.getLore().filter { memberPattern.matches(it) }
-            val memberLevels = members.map { memberPattern.matchEntire(it)?.groupValues?.get(2)?.toInt() ?: 0 }
-            val memberClasses = members.map { memberPattern.matchEntire(it)?.groupValues?.get(1) ?: "" }
 
-            if (memberLevels.any { it <= config.markBelowClassLevel }) {
-                slot highlight LorenzColor.YELLOW
-                continue
-            }
+    }
 
-            if (config.markMissingClass && memberClasses.none { it == selectedClass }) {
-                slot highlight LorenzColor.GREEN
+    @SubscribeEvent
+    fun onToolTipRender(event: LorenzToolTipEvent) {
+        if (!isEnabled()) return
+        if (!inInventory) return
+        val toolTip = toolTipMap[event.slot.slotIndex]
+        if (toolTip.isNullOrEmpty()) return
+        val oldToolTip = event.toolTip
+        for((index, line) in toolTip.withIndex()) {
+            if (index >= event.toolTip.size-1) {
+                event.toolTip.add(line)
+                continue
             }
+            if(oldToolTip[index] != line) event.toolTip[index + 1] = line
         }
```

Comment:
This is to over complicated. But since the backend isn't there I added a todo for me
```suggestion
       // TODO @Thunderblade73 fix that to "event.toolTip = toolTip"
        val oldToolTip = event.toolTip
        for((index, line) in toolTip.withIndex()) {
            if (index >= event.toolTip.size-1) {
                event.toolTip.add(line)
                continue
            }
            if(oldToolTip[index] != line) event.toolTip[index + 1] = line
        }
```